programs/drift/src/state/fulfillment_params/openbook_v2.rs

```diff
@@ -0,0 +1,506 @@
+                AccountMeta::new(*self.openbook_v2_event_heap.key, false),
+                AccountMeta::new(self.base_market_vault.key(), false),
+                AccountMeta::new(self.quote_market_vault.key(), false),
+                AccountMeta::new_readonly(*self.openbook_v2_context.openbook_v2_program.key, false),
+                AccountMeta::new_readonly(*self.openbook_v2_context.openbook_v2_program.key, false),
+                AccountMeta::new_readonly(*self.token_program.key, false),
+                AccountMeta::new_readonly(*self.system_program.key, false),
+                AccountMeta::new_readonly(*self.openbook_v2_context.openbook_v2_program.key, false),
+            ],
+            data,
+        };
+        let account_infos = [
+            self.openbook_v2_context.openbook_v2_program.clone(),
+            self.drift_signer.clone(),
+            self.drift_signer.clone(),
+            self.openbook_v2_context.openbook_v2_market.clone(),
+            self.openbook_v2_market_authority.clone(),
+            self.openbook_v2_bids.clone(),
+            self.openbook_v2_asks.clone(),
+            self.openbook_v2_base_vault.clone(),
+            self.openbook_v2_quote_vault.clone(),
+            self.openbook_v2_event_heap.clone(),
+            self.base_market_vault.to_account_info(),
+            self.quote_market_vault.to_account_info(),
+            self.openbook_v2_context.openbook_v2_program.clone(),
+            self.openbook_v2_context.openbook_v2_program.clone(),
+            self.token_program.to_account_info(),
+            self.system_program.to_account_info(),
+            self.openbook_v2_context.openbook_v2_program.clone(),
+        ];
+        let signer_seeds = get_signer_seeds(&self.signer_nonce);
+        let signers_seeds = &[&signer_seeds[..]];
+
+        invoke_signed_unchecked(
+            &new_place_take_order_instruction,
+            // TODO .to_account_infos()
+            &account_infos,
+            signers_seeds,
+        )
+            .map_err(|e| {
+                msg!("{:?}", e);
+                ErrorCode::FailedOpenbookV2CPI
+            })?;
+
+        Ok(())
+    }
+}
+
+impl<'a, 'b> SpotFulfillmentParams for OpenbookV2FulfillmentParams<'a, 'b> {
+    fn is_external(&self) -> bool {
+        true
+    }
+    fn fulfill_order(&mut self,
+                     taker_direction: PositionDirection,
+                     taker_price: u64,
+                     taker_base_asset_amount: u64,
+                     taker_max_quote_asset_amount: u64,
+    ) -> DriftResult<ExternalSpotFill> {
+        // load openbook v2 market
+        let market = self.openbook_v2_context.load_openbook_v2_market()?;
+        // coin - base
+        // pc - quote
+
+        let serum_max_coin_qty = calculate_serum_max_coin_qty(
+            taker_base_asset_amount,
+            market.base_lot_size as u64,
+        )?;
+
+        let price_lots = calculate_serum_limit_price(
```

Comment:
to confirm serum uses the same type of math?

Reply:
yes